odxtools/diaglayer.py

```diff
@@ -207,10 +193,26 @@ def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
 
         return result
 
-    def _resolve_references(self, odxlinks: OdxLinkDatabase) -> None:
+    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
         """Recursively resolve all references."""
 
-        self.diag_layer_raw._resolve_references(self, odxlinks)
+        self.diag_layer_raw._resolve_odxlinks(odxlinks)
+
+    def _finalize_init(self, odxlinks: OdxLinkDatabase) -> None:
```

Comment:
Not `_resolve_snrefs` like the others?

Reply:
this is the diaglayer. it calls `_resolve_snrefs()` for all its sub-objects...